applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -1026,7 +1026,15 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateAll(MatrixType&        rLe
     const auto integration_coefficients =
         this->CalculateIntegrationCoefficients(IntegrationPoints, Variables.detJContainer);
 
+    auto deformation_gradients = std::vector<Matrix>{};
     for (unsigned int GPoint = 0; GPoint < NumGPoints; ++GPoint) {
+        deformation_gradients.emplace_back(this->CalculateDeformationGradient(GPoint));
+    }
+
+    for (unsigned int GPoint = 0; GPoint < NumGPoints; ++GPoint) {
+        Variables.F    = deformation_gradients[GPoint];
+        Variables.detF = MathUtils<>::Det(Variables.F);
```

Comment:
In the original version, F and detF were not recalculated here. Please could you explain the reason?

Reply:
I have reverted this change, since it didn't replace another operation yet. I agree with you that if we change it here, something else somewhere else should also change. Otherwise, we are simply doing more work without any benefits.